packages/console/src/components/ReportConversion/utils.ts

```diff
@@ -37,8 +37,45 @@ export function lintrk(..._: unknown[]) {
     window.lintrk = { q: [] };
   }
 
-  // eslint-disable-next-line prefer-rest-params
   window.lintrk.q.push(arguments);
 }
 
-/* eslint-enable @silverhand/fp/no-mutation, @silverhand/fp/no-mutating-methods */
+/**
+ * This function will do the following things:
+ *
+ * 1. Canonicalize the given email by Reddit's rule: lowercase, trim,
+ * remove dots, remove everything after the first '+'.
+ * 2. Hash the canonicalized email by SHA256.
+ */
+export const hashEmail = async (email?: string) => {
+  if (!email) {
+    return;
+  }
+
+  const splitEmail = email.toLocaleLowerCase().trim().split('@');
+  const [localPart, domain] = splitEmail;
+
+  if (!localPart || !domain || splitEmail.length > 2) {
+    return;
+  }
```

Comment:
Nit: should we use a regex to strictly validate the email?

Reply:
i think it's fine, since `primaryEmail` is from us, and in most cases the address will come from us. plus, it's for tracking purpose, even if the email is not valid, as long as it can be recognized after hash then it should good